library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/GenericWireframeMapper.kt

```diff
@@ -28,10 +28,26 @@ internal abstract class GenericWireframeMapper(
     private val decorViewMapper: DecorViewMapper,
     private val checkBoxMapper: CheckBoxMapper,
     private val radioButtonMapper: RadioButtonMapper,
-    private val switchCompatMapper: SwitchCompatMapper
+    private val switchCompatMapper: SwitchCompatMapper,
+    private val customMappers: Map<Class<*>, WireframeMapper<View, *>>
 ) : WireframeMapper<View, MobileSegment.Wireframe> {
 
     override fun map(view: View, systemInformation: SystemInformation):
+        List<MobileSegment.Wireframe> {
+        return applyCustomMappers(view, systemInformation)
```

Comment:
minor: wouldn't it be simpler to just have a list of custom + default mappers (in this order) and just return `firstNotNull` while iterating over it? but probably it doesn't bring a lot of added value.

Reply:
yeah good idea...I am going to give it a try.